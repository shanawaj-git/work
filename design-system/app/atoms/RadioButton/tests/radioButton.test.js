import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { render } from 'react-testing-library';
import RadioButtonComponentIndex from '../index';

const shallowRenderer = new ShallowRenderer();

describe('RadioButton.js', () => {
  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = render(
        <RadioButtonComponentIndex
          groupLabel="Gender"
          row
          values={{ Male: 'boy', Female: 'girl' }}
          disabled={false}
        />,
      );
      expect(tree.container).toBeTruthy();
    });

    it('Should display RadioButton with primary color style', () => {
      shallowRenderer.render(
        <RadioButtonComponentIndex
          groupLabel="Gender"
          row
          values={{ Male: 'boy', Female: 'girl' }}
          disabled={false}
        />,
      );
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should display disabled RadioButton', () => {
      shallowRenderer.render(
        <RadioButtonComponentIndex
          groupLabel="Gender"
          row
          disabled
          values={{ Male: 'boy', Female: 'girl' }}
        />,
      );
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });
  });
});
