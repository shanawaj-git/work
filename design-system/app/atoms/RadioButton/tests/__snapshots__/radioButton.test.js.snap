// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton.js index Should display RadioButton with primary color style 1`] = `
<ForwardRef(Styled(Component))
  disabled={false}
  groupLabel="Gender"
  row={true}
  values={
    Object {
      "Female": "girl",
      "Male": "boy",
    }
  }
/>
`;

exports[`RadioButton.js index Should display disabled RadioButton 1`] = `
<ForwardRef(Styled(Component))
  disabled={true}
  groupLabel="Gender"
  row={true}
  values={
    Object {
      "Female": "girl",
      "Male": "boy",
    }
  }
/>
`;
